src/transformers/pipelines/automatic_speech_recognition.py

```diff
@@ -30,18 +30,33 @@
     from ..models.auto.modeling_auto import MODEL_FOR_CTC_MAPPING, MODEL_FOR_SPEECH_SEQ_2_SEQ_MAPPING
 
 
```

Comment:
tl;dr of changes to this function:

This function takes a list of strides `stride`. It previously took a single number `ratio`. Now, it takes a list of numbers `ratio` to match the list of strides. This lets the ratio differ between strides.